src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -25,7 +25,7 @@
  */
 final class Dotenv
 {
-    public const VARNAME_REGEX = '(?i:[A-Z][A-Z0-9_]*+)';
+    public const VARNAME_REGEX = '(?i:[A-Z_][A-Z0-9_]*+)';
```

Comment:
This new regexp makes `_` a valid env var, which is wrong. Can you fix the regex and add a test to make sure that `_` is not allowed?

Reply:
This should do the trick, except the mentioned double quotes issue:
https://regex101.com/r/BK7AN6/1

Reply:
this should be fixed now